inquire/src/prompts/path_select/mod.rs

```diff
@@ -0,0 +1,503 @@
+mod action;
+use action::*;
+mod prompt;
+use prompt::*;
+mod config;
+use config::*;
+
+use crate::{
+    config::get_configuration,
+    error::InquireResult,
+    formatter::MultiOptionFormatter,
+    list_option::ListOption,
+    prompts::prompt::Prompt,
+    terminal::get_default_terminal,
+    type_aliases::Filter,
+    ui::{Backend, MultiSelectBackend, RenderConfig},
+    InquireError,
+};
+use std::{
+    convert::TryFrom,
+    ffi::OsStr,
+    fmt, fs,
+    ops::Deref,
+    path::{Path, PathBuf},
+};
+
+/// Different path selection modes specify what the user can choose
+#[derive(Clone, Eq, PartialEq)]
+pub enum PathSelectionMode<'a> {
+    /// The user may pick a file with the given (optional) extension
+    File(Option<&'a str>),
```

Comment:
@LeoRiether, is using `Cow<'a, str>` here a good idea? From your comments on that another issue

Reply:
There are pros and cons. `Cow` in some cases does provide easier lifetime handling, but it also adds some complexity to the API, like the need to write `File(Some("jpg".into()))` instead of `File(Some("jpg"))` and some users saying "what the hell is a Cow?"

Maybe it's worth it? I didn't find clear guidelines about this online, but I'll look into it again later to try and come up with a more definitive answer :)

Reply:
Also, I feel like it would be nice to have a more flexible `PathSelectionMode`. I don't see an easy way, for example, to integrate the [ignore](https://crates.io/crates/ignore) crate to list files respecting `.gitignore`. If I have any ideas of how to actually implement this more flexible API, I'll make sure to post them.

Reply:
> . I don't see an easy way, for example, to integrate the [ignore](https://crates.io/crates/ignore) crate to list files respecting `.gitignore`.

Hiya. I'm not sure what you're asking for here. Would it help if I tried to add globs or some kind of name-matching/name-excluding patterns? I looked at the `ignore` crate, but I don't know if adding that dependency would be better than just using the `regex` crate. Are you wanting to respect `.gitignore` files in the ancestry of paths too?